### ISSUE TITLE
[`pylint`] Extend `PLW0133` to also detect user-defined exceptions

### ISSUE THREAD

**Issue by LoicRiegel** [astral-sh/ruff#17347]
### Summary

From the [rule's documentation](https://docs.astral.sh/ruff/rules/useless-exception-statement/#useless-exception-statement-plw0133) (PLW0133):

> Known problems
>
> This rule only detects built-in exceptions, like ValueError, and does not catch user-defined exceptions.

So currently this is caught:
```py
def foo():
    ValueError("...")
```
But not this. But I would love if it would, it would have spared me 1hr of debugging this morning :)
```py
class MyCustomError(Exception):
    ...

def foo():
    MyCustomError("...")
```
